src/components/SingleArticlePage/PostComment.jsx

```diff
@@ -12,54 +12,60 @@ export const PostComment = ({ commentsList, setCommentsList }) => {
   const [userLoggedIn, setUserLoggedIn] = useState(true);
   const { articleId } = useParams();
   const [err, setErr] = useState(null);
+  const [isSubmitting, setIsSubmitting] = useState(false);
 
   useEffect(() => {
-    if (currentUser !== null) {
+    if (currentUser.username !== undefined) {
       setUserLoggedIn(true);
       setComment((currComment) => ({
         ...currComment,
         username: currentUser.username,
       }));
-    } else {
-      setUserLoggedIn(false);
     }
   }, [currentUser]);
 
   const handleSubmit = (e) => {
     e.preventDefault();
-    const copy = { ...comment };
-    copy.author = comment.username;
-    copy.result = 1;
-    copy.total_results = commentsList[0].total_results;
-    copy.comment_id = Date.now();
-    copy.created_at = Date.now();
-    copy.userComment = true;
-    copy.tempComment = true;
-    setCommentsList([copy, ...commentsList]);
-    postComment(articleId, comment)
-      .then((response) => {
-        const postedComment = response.comment;
-        setCommentsList((currCommentsList) => {
-          postedComment.result = 1;
-          postedComment.total_results =
-            Number(commentsList[0].total_results) + 1;
-          const filteredComments = currCommentsList.filter(
-            (currComment) => !currComment.tempComment
-          );
-          filteredComments.forEach((comment) => {
-            comment.result = Number(comment.result) + 1;
-            comment.total_results = Number(comment.total_results) + 1;
-          });
-          const copy = [postedComment, ...filteredComments];
+    if (currentUser.username === undefined) {
+      setUserLoggedIn(false);
+    } else {
+      setIsSubmitting(true);
+      const copy = { ...comment };
+      copy.author = comment.username;
```

Comment:
We don't want to optimistically render the posting of comments as it means the ID is not legitimate - this can cause errors. For this you would ideally display some user feedback that the post is in progress, then once the post has completed (in the .then block) you would update your comment state with the returned posted object